### DISCUSSION TITLE
Integrating next-video with Next.js Configuration in ESM Format

### DISCUSSION THREAD

**Discussion by jensjohnston** [muxinc/next-video#183] [Q&A] [answered]
Hello everyone,

I'm currently working on a Next.js project and trying to integrate next-video into my next.config.mjs file. My project uses ES Module (ESM) syntax, and I'm facing difficulties adapting the CommonJS syntax used by next-video into ESM.

Here's the snippet I'm trying to integrate:

```
const { withNextVideo } = require('next-video/process');
const nextConfig = {}; // Your current Next Config object
module.exports = withNextVideo(nextConfig);
```

Given that my next.config.mjs file uses ESM syntax (as shown below), I'm unsure how to properly import and use withNextVideo from next-video

```
import someOtherModule from 'some-other-module';

const nextConfig = {
  // My current Next.js configuration
};

export default nextConfig;
```

1. Does next-video support ESM, and if so, how can I import it correctly in my next.config.mjs file?
2. If next-video does not support ESM directly, is there a recommended workaround to use it within an ESM context?
3. Are there any best practices or examples of integrating CommonJS modules into ESM-configured Next.js projects that I should be aware of?

**Comment by luwes** [comment, marked as answer]
Thanks for the question!

1. Yes, next-video supports ESM.

Here is an example of how your next.config.mjs could look like.

```js
import { withNextVideo } from "next-video/process";

const nextConfig = {
  // My current Next.js configuration
};

export default withNextVideo(nextConfig);
```
